frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.tsx

```diff
@@ -542,15 +542,7 @@ function renderOptions({
   if (alwaysShowOptions || isFocused) {
     if (optionsList) {
       return optionsList;
-    } else if (
-      hasList({
-        parameter,
-        fields,
-        disableSearch,
-        options,
-      }) &&
-      valuesMode === "list"
-    ) {
+    } else if (!_.isEmpty(options) && valuesMode === "list") {
```

Comment:
We can do this replacement because `valuesMode === "list"` implies `shouldList(...)`, and 
`hasList(...) iff (!_.isEmpty(options) AND shouldList(...))`